core/src/block.rs

```diff
@@ -81,9 +81,10 @@ impl Chain {
         ChainIterator::new(self)
     }
 
-    /// Latest block reference and its height.
-    pub fn latest_block(&self) -> Option<MapRef<u64, VersionedCommittedBlock>> {
-        self.blocks.get(&(self.blocks.len() as u64))
+    /// Get block nth from the end of the [`Chain`]
+    pub fn latest_nth_block(&self, n: usize) -> Option<MapRef<u64, VersionedCommittedBlock>> {
+        self.blocks
+            .get(&(self.blocks.len().saturating_sub(n) as u64))
```

Comment:
I think `saturating_sub` behaviour is non-obvious and should be either documented, or changed to return None when n > len

Reply:
Maybe also rename `latest_nth_block` and friends to `nth_back` by analogue to `std::iter::DoubleEndedIterator`?

Reply:
> `nth_back`
Thanks, sounds much better! 

> I think saturating_sub behaviour is non-obvious and should be either documented, or changed to return None when n > len

I will add docs

Reply:
> Will return `None`

Oh, right, didn't realize block height starts from 1.